growattServer/__init__.py

```diff
@@ -521,32 +876,45 @@ def plant_info(self, plant_id):
         """
         Get basic plant information with device list.
         """
-        response = self.session.get(self.get_url('newTwoPlantAPI.do'), params={
+        response = self.session.get(self.__get_url('newTwoPlantAPI.do'), params={
             'op': 'getAllDeviceListTwo',
             'plantId': plant_id,
             'pageNum': 1,
             'pageSize': 1
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to get plant info')
 
-    def get_plant_settings(self, plant_id):
+    def get_all_devices(self, plant_id):
         """
-        Returns a dictionary containing the settings for the specified plant
+        Get basic plant information with device list.
+        """
+        response = self.session.get(self.__get_url('newTwoPlantAPI.do'), 
+                                     params={'op': 'getAllDeviceList',                                
+                                             'plantId': plant_id,
+                                             'language': 1})
 
-        Keyword arguments:
-        plant_id -- The id of the plant you want the settings of
+        if response.status_code == 200:
+            return response.json()['deviceList']
+        else:
+            raise Exception('Failed to get device data')
```

Comment:
Does this add anything more than
https://github.com/indykoning/PyPi_GrowattServer/blob/75b6b73d3aa7f4014ce6759a3067d5c1375e74f3/growattServer/__init__.py#L514

And if so is it backwards compatible with additional data (or minor adjustment)?
If we can get it to return the same as the original function we can replace it.
I'd rather have that than "duplicate" calls which might make it confusing which to use.

Reply:
The reason I created get_all_devices() is because device_list() does not return anything for my system, just an empty dict. Maybe get_all_devices() could replace device_list(), but I have no way of testing this on all other systems so it could very well be that get_all_devices() would return empty on non tlx system. 

I see two options: 

1. having one public wrapper function that checks some plant info and calls the two different APIs depending on what type of system it is. 
2. Having two functions device_list() and tlx_device_list().

If we go with 1, one could check "deviceType" of "invList". This is what my invList looks like:

"invList": [
        {
            "bMerterConnectFlag": 0,
            "bdc1Soc": 13,
            "bdc2Soc": 0,
            "datalogSn": “XXXX”,
            "deviceAilas": “XXXX”,
            "deviceSn": “XXXXX,
            "deviceStatus": "0",
            "deviceType": "tlx",
            "eToday": "24.8",
            "eTodayStr": "24.8kWh",
            "energy": "3720.5",
            "location": "",
            "lost": false,
            "plantId": 2450229,
            "power": "0",
            "powerStr": "0kW",
            "prePto": "-1",
            "type": 0,
            "type2": "6"
        }

Let me know what you think.